lib/perm/can-order-for.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+var _ = require('lodash');
+
+
+function canOrderFor (args, cb) {
+  const seneca = this;
+  const plugin = args.role;
+  const userId = args.user.id;
+  const applications = args.params.applications;
+
+  seneca.act({role: 'cd-profiles', cmd: 'load_children_for_user', userId: userId, user: args.user },
+    function(err, children) {
+      const childIds = children.map(child => child.userId);
+
+      const invalidApplications = applications.filter(application => (userId !== application.userId && childIds.indexOf(application.userId) === -1));
```

Comment:
Shouldn't that be an `||` rather than an `&&` ? Self-booking becomes impossible if you don't append the current userId to the list of childIds/allowedIds

Reply:
We went through this a few times before settling on the answer of `&&`. We want this to result in an array of invalid applications. In order for an application to be invalid, it has to satisfy both criteria.

If we used `||` then an application for yourself would return true because you're not one of your children, and an application for your child would also return true because your child isn't you.

With `&&` it will return false for an application for yourself, and it will return false for an application for your child. That's what we want, so we get an array of invalid applications.

Reply:
I think the issue is that the function is "can-book-for" which test a validity scenario, but we're looking for "invalidity" rather than validity. Hence the "mental gym" to do (positive scenario vs negative one) the translation from one "positive function" to a negative one.
Would it be easier to understand if we were checking the length of valid ones compared to the original one ?

Reply:
Fair point. I'd be up for that.

Reply:
@heyitsmaimai Does it make sense? :)